test/regressions/vite.config.mts

```diff
@@ -0,0 +1,85 @@
+import { defineConfig, transformWithEsbuild } from 'vite';
+import react from '@vitejs/plugin-react';
+import * as path from 'path';
+import * as url from 'url';
+
+const currentDirectory = url.fileURLToPath(new URL('.', import.meta.url));
+const WORKSPACE_ROOT = path.resolve(currentDirectory, '../../');
+
+// https://vite.dev/config/
+export default defineConfig({
+  plugins: [
+    {
+      // Unfortunately necessary as we opted to write our jsx in js files
+      name: 'treat-js-files-as-jsx',
+      async transform(code, id) {
+        if (/\/node_modules\//.test(id)) {
+          return null;
+        }
+        if (!/.*\.js$/.test(id)) {
+          return null;
+        }
+        if (id.startsWith('\0')) {
+          return null;
+        }
+        // Use the exposed transform from vite, instead of directly
+        // transforming with esbuild
+        return transformWithEsbuild(code, id, {
+          loader: 'tsx',
+          jsx: 'automatic',
+        });
+      },
+    },
+    react(),
+  ],
+  define: {
+    'process.env': '{}',
+    DISABLE_CHANCE_RANDOM: JSON.stringify(true),
+  },
+  resolve: {
+    alias: {
+      '@mui/docs': path.resolve(
+        WORKSPACE_ROOT,
+        './node_modules/@mui/monorepo/packages/mui-docs/src',
+      ),
+      '@mui/x-data-grid': path.resolve(WORKSPACE_ROOT, './packages/x-data-grid/src'),
+      '@mui/x-data-grid-generator': path.resolve(
+        WORKSPACE_ROOT,
+        './packages/x-data-grid-generator/src',
+      ),
+      '@mui/x-data-grid-pro': path.resolve(WORKSPACE_ROOT, './packages/x-data-grid-pro/src'),
+      '@mui/x-data-grid-premium': path.resolve(
+        WORKSPACE_ROOT,
+        './packages/x-data-grid-premium/src',
+      ),
+      '@mui/x-date-pickers': path.resolve(WORKSPACE_ROOT, './packages/x-date-pickers/src'),
+      '@mui/x-date-pickers-pro': path.resolve(WORKSPACE_ROOT, './packages/x-date-pickers-pro/src'),
+      '@mui/x-charts': path.resolve(WORKSPACE_ROOT, './packages/x-charts/src'),
+      '@mui/x-charts-pro': path.resolve(WORKSPACE_ROOT, './packages/x-charts-pro/src'),
+      '@mui/x-charts-vendor': path.resolve(WORKSPACE_ROOT, './packages/x-charts-vendor/es'),
```

Comment:
Without `es` it targets CJS and fails to resolve imports.

Reply:
I'm not a fan of introducing yet another list of all the packages
I feel like we already have too many of those, and each new one is a new small pain point when creating a package.
But not sure if we can achieve the same result without it :/

Reply:
In #17214 I've gotten rid of path aliases in a lot of places. The rationale is that we should test the code as it is seen by users, which means we should `pnpm link` the built folders rather than alias to local paths. It has highlighted ESM compat issues that we wouldn't have seen otherwise.

Reply:
> But not sure if we can achieve the same result without it :/

https://www.npmjs.com/package/vite-tsconfig-paths

Reply:
Great points Rom. 👍 
I think those discussion points should be kept on your PR. 👌 
Your suggestion makes sense to me, but I'd love to hear feedback from others. 😃